src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -174,6 +175,19 @@ static ucs_status_t uct_cuda_ipc_rkey_release(uct_component_t *component,
                                               uct_rkey_t rkey, void *handle)
 {
     ucs_assert(NULL == handle);
+
+    uct_cuda_ipc_key_t *key = (uct_cuda_ipc_key_t *) rkey;
+    if (key->unmap_memhandle_func != 0 && (void *)key->d_mapped != 0) {
+        ucs_trace("unmapping addr: %p", (void *)key->d_mapped);
+        ucs_status_t status = key->unmap_memhandle_func((void *)key->d_mapped);
+        if (status != UCS_OK) {
+            ucs_fatal("failed to unmap addr: %p", (void *)key->d_mapped);
+        }
+    } else {
+        ucs_trace("unmap_memhandle_func not defined, cannot unamp addr: %p",
```

Comment:
is not it an error flow? maybe use `ucs_error` or `ucs_warn` then?

Reply:
No, that is only defined when `UCX_CUDA_IPC_CACHE=n`, otherwise the cache should take care of the unmapping.

Reply:
Since that may be misleading when `UCX_CUDA_IPC_CACHE=y` I removed that logging message.